e2e_example/test/build_integration_test.dart

```diff
@@ -75,4 +76,30 @@ void main() {
       expect(result.stdout, contains('Skipping tests due to build failure'));
     });
   });
+
+  group('regression tests', () {
+    test('Failing optional outputs which are required during the next build',
+        () async {
+      // Run a build with DDC that should fail
+      final path = p.join('test', 'common', 'message.dart');
+      await replaceAllInFile(path, "'Hello World!'", 'can not parse this');
+      final result = await runAutoBuild(trailingArgs: ['--fail-on-severe']);
+      expect(result.exitCode, isNot(0));
+      expect(result.stderr, contains('Failed'));
+
+      // Run a build with dart2js that should also fail. Makes the failing DDC
```

Comment:
Not super stoked on relying on the internals here... also note that the ddc output is going to get marked as needing an update as well which might change the logic or hit different codepaths (not sure how we treat failures that didn't rerun).

I would instead create a new file with an error in it (but not imported anywhere) and expect as successful build, then add an import to it and expect a failure, then remove the import again and expect it to pass.

Reply:
As a bonus it's also much faster :)

Skipping the check for the successful build since I don't think it changes the meaning of the test.